apps/sms/views/inbox/js/inbox.js

```diff
@@ -806,29 +708,16 @@ var InboxView = {
     Threads.delete(threadId);
 
     // Cleanup the DOM
-    this.removeConversationDOM(threadId);
+    var index = this.listData.findIndex((data) => data.id === threadId);
+    if (index > -1) {
+      this.listData.splice(index, 1);
+      this.list.setModel(this.listData);
```

Comment:
> Sorry I'm not very familiar with RxJS, but the methods provide by datasource seems more powerful than native array, and the key point is you can centralize the data changes and using throttle to handle all the setModel/rendering. It would be more clean and readable. Maybe I can try to find some other native approach to reach this idea.

Yeah, it just was rough idea inspired by RxJS, but I think we need single place to set model list model that can probably support throttling as well, otherwise we'll have to spread ```setModel``` calls across the entire view